packages/@tinacms/graphql/src/primitives/resolver/index.ts

```diff
@@ -644,7 +644,9 @@ const resolveDateInput = (
   /**
    * Convert string to `new Date()`
    */
-  const date = parseISO(value)
+  const parsedValue = new Date(value).toISOString()
```

Comment:
This seems redundant, as `parseISO` returns a date object as well. So here we're creating a date object, converting it to ISO, and then parsing it from ISO back into a date.

If we want to proceed with this change, we could accomplish the same with `const date = new Date(value)` but I'm concerned that this could just be masking the problem by using a more generous parser

Reply:
I was thinking about this more last night after I pushed this up.

That check was the first fix to handle the untouched date field, since the value in that situation would come through as `"July 12, 2021"` for example, thus hitting the invalid date error.

Probably not the best solution but this would at least take that string, and convert to a date object before going through the ISO parse.

I'm sure there is a better way to address.

Reply:
The purpose of lines 647 & 648 here are to create a date object. The proposed change parses the passed-in value using the Date constructor's parsing method (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse), converts the resulting `Date` object into an ISO string, and then parses that ISO string back into a `Date` object. That's what I meant about it being redundant.

If we think that using `Date.parse` instead of `parseISO` is the right solution to the problem, then we can do that more explicitly without passing through the arbitrary ISO conversion. But that has the feel of trying to address a symptom rather than the cause, since the frontend should be able to pass whatever format the backend needs

Reply:
And this is a problem because we're literally sending in the string `Jul 2, 2020` right? And we can't enforce it to be UTC because final form doesn't call parse on that field unless it's touched. Do I have that right? If that's the case, the best course of action may be to require UTC format from the UI and not allow another format

Reply:
It sounds like the proper fix is for the input the field to have its own local state of the _formatted_ version of the time, and final form only knows about the UTC version.

Reply:
this seems related https://github.com/final-form/react-final-form/issues/431

Reply:
Yeah, that issue seems to indicate that they don't have a concept of presentation vs value, which leads me to think we need to manage the presentation ourselves with local state